stl/inc/cmath

```diff
@@ -1335,6 +1337,40 @@ _NODISCARD auto hypot(const _Ty1 _Dx, const _Ty2 _Dy, const _Ty3 _Dz) {
 }
 
 #if _HAS_CXX20
+template <class _Ty>
+inline constexpr _Ty _Floating_max;
+
+template <>
+inline constexpr float _Floating_max<float> = FLT_MAX;
+
+template <>
+inline constexpr double _Floating_max<double> = DBL_MAX;
+
+template <>
+inline constexpr long double _Floating_max<long double> = LDBL_MAX;
+
+template <class _Ty>
+_NODISCARD constexpr _Ty _Linear_for_lerp(const _Ty _ArgA, const _Ty _ArgB, const _Ty _ArgT) {
+    if (_STD is_constant_evaluated()) {
+        auto _Smaller     = _ArgT;
+        auto _Larger      = _ArgB - _ArgA;
+        auto _Abs_smaller = _Float_abs(_Smaller);
+        auto _Abs_larger  = _Float_abs(_Larger);
+        if (_Abs_larger < _Abs_smaller) {
+            _STD swap(_Smaller, _Larger);
+            _STD swap(_Abs_smaller, _Abs_larger);
+        }
+
+        if (_Abs_smaller > 1 && _Abs_larger > _Floating_max<remove_cv_t<_Ty>> / _Abs_larger) {

```

Comment:
I see that @MattStephanson in https://github.com/microsoft/STL/pull/1918#issuecomment-841042608 originally suggested
```cpp
if (_Abs_larger > _STD sqrt(numeric_limits<_Ty>::max()) && _Abs_smaller > 1) {
```
so this is a rephrasing that avoids `sqrt`, but it still seems to me that
```cpp
if (_Abs_smaller > 1 && _Abs_larger > _Floating_max<remove_cv_t<_Ty>> / _Abs_smaller) {
```
would be better to test here. Again, I could be wrong.

Reply:
Yes, you are correct. That will activate this codepath more often than necessary but it is probably OK, because the two path should give the same result when no overflow.
I think it was just a slightly more rough estimate and not typo.
But I think you are right and we can replace it with `_Abs_smaller`.
But just in case, I will also call @MattStephanson , if you have time could you take a look?

Reply:
The test suggested by @StephanTLavavej is more precise and appears to be at least mostly correct (it might be possible that `_Abs_larger` is slightly larger than the exact quotient, but the the quotient gets rounded up to be equal to `_Abs_larger` and the fast path is mistakenly taken.) Nonetheless, I have concern about the test. (1) This is the `constexpr` path, so performance is less critical; (2) the `sqrt(max)` could be precalculated; and (3) I doubt that always doing 1 division to sometimes save 3 multiples is a good tradeoff.

Reply:
I like the idea because now we can remove the additional include files.

Reply:
> Can/should we simplify the condition to just _Abs_smaller > 1?

FWIW, I think this is safe. You're correct in pointing out that there are two separate issues, performance and correctness, and the correctness of the `0.5`-scaled path depends on not underflowing. I think that ensuring that both multiplicands are at least 1 covers that.